### PR TITLE
Handle always included [de]activation scripts

### PR DESCRIPTION
**Pull request by jakirkham**
Tries to handle the now always included [de]activation scripts.

cc @pelson @bollwyvl @msarahan

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,8 +15,14 @@ build:
   #
   # Note: The Windows copy of openssl does not include these.
   always_include_files:
-    - ssl/cert.pem      # [unix]
-    - ssl/cacert.pem    # [unix]
+    - ssl/cert.pem                                                       # [unix]
+    - ssl/cacert.pem                                                     # [unix]
+    - Library\\ssl\\cert.pem                                             # [win]
+    - Library\\ssl\\cacert.pem                                           # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: I really don't understand how this still isn't needed for Windows, but it appears it still isn't.

**Comment by msarahan** [comment]
No library in win path, just prefix:

https://github.com/conda/conda/blob/master/bin/activate.bat#L76

**Comment by jakirkham** [comment]
> I won't pretend to fully understand it, but it has gone green!

It is fortunately not that bad. As I mentioned to @msarahan earlier, the problem is we need to use `python` and `certifi` as build dependencies. So, we have this chain `certifi` -> `python` -> `openssl` -> `ca-certificates`. As a result, we find ourselves depending on the last build of `ca-certificates` when we build the next one.

I've added some tests to make sure it works. Though I think it is my bad not adding them before. 😞

> If you bump the build number, I don't have any objections 👍

I can do that, but only because I deleted the old build number. If I didn't do that, we would be in the jackpot w.r.t. `always_include_files` as we would be missing the old [de]activate scripts. Though @msarahan is softening that constraint so it doesn't fail to build if they are missing. So ideally we won't have to think about that problem much longer.

**Comment by jakirkham** [comment]
> No library in win path, just prefix:

> https://github.com/conda/conda/blob/master/bin/activate.bat#L76

Ok, we can fix that. Though this is a holdover from the PR that added the activation scripts. See this [line]( https://github.com/conda-forge/ca-certificates-feedstock/pull/2/files#diff-d04c86b6bb20341f5f7c53165501a393R17 ).

In any event, it still doesn't explain why `always_include_files` doesn't pick them up. They are present in that Windows package.

**Comment by jakirkham** [comment]
I think I have addressed the new round of comments. Have tested this locally with `conda` 4.1.6 on my Mac and can verify it works. Can't say anything about Windows though.

**Comment by jakirkham** [comment]
Removing WIP as this is done on my end. Please let me know if needs anything else @msarahan or @pelson.

File: recipe/activate.bat [hunk 2]
```diff
@@ -1,2 +1,3 @@
-set "SSL_CERT_DIR=%CONDA_ENV_PATH%\ssl"
+:: handle conda...   >=4.1.3        <= 4.1.2
+set "SSL_CERT_DIR=%CONDA_PREFIX%%CONDA_ENV_PATH%\ssl"
```
Comment (jakirkham) [inline recipe/activate.bat]: Maybe this works in Windows? If not this, we may need to do something else.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -7,16 +7,22 @@ package:
...
   always_include_files:
-    - ssl/cert.pem      # [unix]
-    - ssl/cacert.pem    # [unix]
+    - ssl/cert.pem                                                       # [unix]
+    - ssl/cacert.pem                                                     # [unix]
+   #- Library\\ssl\\cert.pem                                             # [win]
+   #- Library\\ssl\\cacert.pem                                           # [win]
+    - etc/conda/activate.d/ca-certificates_activate.sh                   # [unix]
+    - etc/conda/deactivate.d/ca-certificates_deactivate.sh               # [unix]
+   #- etc\\conda\\activate.d\\ca-certificates_activate.bat               # [win]
```
Comment (msarahan) [inline recipe/meta.yaml]: Can we re-enable these?

Comment (msarahan) [inline recipe/meta.yaml]: I guess you're waiting on the conda-build update that allows your build to succeed.  It'll be up soon.

Comment (jakirkham) [inline recipe/meta.yaml]: Yep. Now it's switched though.

**Comment by jakirkham** [comment]
This is a pretty weird error on AppVeyor. See the snippet below from this [log]( https://ci.appveyor.com/project/conda-forge/ca-certificates-feedstock/build/1.0.15/job/41qwxmblr1wfqx5u#L282 ). It's like it is using the wrong activate script. However, I don't think we are including it.

```
C:\conda\conda-bld\test-tmp_dir>activate _test 
The system cannot find the file specified.
'export' is not recognized as an internal or external command,
operable program or batch file.
'export' is not recognized as an internal or external command,
operable program or batch file.
The system cannot find the path specified.
TESTS FAILED: ca-certificates-2016.2.28-3
Command exited with code 1
```

**Comment by jakirkham** [comment]
This happened with `conda-build` version 1.21.5 and 1.21.4 (though the activation syntax changed slightly) as evidenced by this [log]( https://ci.appveyor.com/project/conda-forge/ca-certificates-feedstock/build/1.0.15/job/41qwxmblr1wfqx5u#L282 ). Not sure what I'm doing wrong, but it points to the wrong activation script being copied.

File: recipe/bld.bat [hunk 4]
```diff
@@ -10,9 +10,9 @@ if errorlevel 1 exit 1
...
 if errorlevel 1 exit 1
 
-:: Copy the [de]activate scripts to %LIBRARY_PREFIX%\etc\conda\[de]activate.d.
+:: Copy the [de]activate scripts to %PREFIX%\etc\conda\[de]activate.d.
 :: This will allow them to be run on environment activation.
 FOR %%F IN (activate deactivate) DO (
-    IF NOT EXIST %LIBRARY_PREFIX%\etc\conda\%%F.d MKDIR %LIBRARY_PREFIX%\etc\conda\%%F.d
-    COPY %RECIPE_DIR%/%%F.bat %LIBRARY_PREFIX%\etc\conda\%%F.d\%PKG_NAME%_%%F.bat
+    IF NOT EXIST %PREFIX%\etc\conda\%%F.d MKDIR %PREFIX%\etc\conda\%%F.d
+    COPY %RECIPE_DIR%/%%F.bat %PREFIX%\etc\conda\%%F.d\%PKG_NAME%_%%F.bat
```
Comment (jakirkham) [inline recipe/bld.bat]: Looks like we are copying a `.bat` file to me. 😕

File: recipe/activate.bat (hunk 2 above)
Comment (jakirkham) [inline recipe/activate.bat]: This doesn't have any `export` statements AFAICT.

**Comment by jakirkham** [comment]
The last successful build used at `conda-build` 1.21.4 too.

**Comment by msarahan** [comment]
yeah, not your problem.  I think the code in conda-build needs to include the extension (or not, as appropriate.)  Will put this into another 1.21 release.

**Comment by jakirkham** [comment]
Maybe there is a `conda-build` bug? IDK.

The only difference I see between the last working and first failing build are these [changes]( https://github.com/conda-forge/ca-certificates-feedstock/compare/cb4fe08e7f4df9102c889ab0be75145b90c1b9e8...5aac6d25fc4b2f84d2e433ca024d975a7b4836a1 ). The versions of everything between this [passing build]( https://ci.appveyor.com/project/conda-forge/ca-certificates-feedstock/build/1.0.12/job/wsx4b9jq5wli1vqb ) and this [failing build]( https://ci.appveyor.com/project/conda-forge/ca-certificates-feedstock/build/1.0.14/job/kki5gtywi3ffi28p ) look identical to me.

**Comment by jakirkham** [comment]
Alright, there is one subtle difference I do see in dependencies that I missed before. We have a build number bump (`0` -> `1`) in `conda-forge`'s `python` version 3.5.2. Not sure how that affects anything, but it seems worth noting.

**Comment by jakirkham** [comment]
So it sounds like we are waiting for PR ( https://github.com/conda/conda-build/pull/1107 ) to make it into a release. Though maybe we should test it here to make sure it fixes the issue.

Is there anything we can do to get this in for the meantime? If not, we should pull build number 1 as that causes issues in some cases due to the very problem that @bollwyvl was trying to fix.

**Comment by 183amir** [comment]
A new conda build was just released.

**Comment by jakirkham** [comment]
Restarted AppVeyor. Let's see what happens.

**Comment by msarahan** [comment]
Dang, this queue is pretty crazy.  I'm itching to know if the new Conda Build fixes things...

**Comment by jakirkham** [comment]
😭 

```
The system cannot find the file specified.
'export' is not recognized as an internal or external command,
operable program or batch file.
'export' is not recognized as an internal or external command,
operable program or batch file.
The system cannot find the path specified.
```

**Comment by jakirkham** [comment]
I have labeled build number 1 as broken so it shouldn't interfere any more. Sorry @183amir. Seems we need a bit more time to figure this out.

**Comment by msarahan** [comment]
Rats.  I'll investigate this more tomorrow.  I'm surprised that this didn't work.

**Comment by jakirkham** [comment]
No worries. This activation stuff (particularly on Windows) seems to be a tricky business. I'm glad to see `conda` and `conda-build` getting smarter about it though.

**Comment by jakirkham** [comment]
So the main motivation (for me at least) to pursue this is gone. Therefore am closing this out.

**Pull request closed without merging**